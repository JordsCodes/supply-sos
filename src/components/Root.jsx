import React from "react";
import { Link, Outlet, useLocation } from "react-router-dom";
import "../styles/root.css";

function Root({ index, setIndex }) {
  const location = useLocation();
  const url = location.pathname;

  const prompts = [
    "/",
    "/year-group",
    "/subject",
    "/topic",
    "/curriculum-aim",
    "/lesson-plan",
  ];

  const goForward = () => {
    setIndex(index + 1);
  };

  const goBack = () => {
    setIndex(index - 1);
  };

  const goHome = () => {
    setIndex(0);
  };

  return (
    <div className="root">
      <h1 className="heading">SupplySoS</h1>
      <div id="prompt">
        <Outlet />
      </div>
      <div className="navbar">
        {url === "/" && (
          <Link
            to="year-group"
            className="button"
            type="button"
            onClick={goForward}
          >
            Let&apos;s get started
          </Link>
        )}
        {(url === "/year-group" ||
          url === "/subject" ||
          url === "/topic" ||
          url === "/curriculum-aim") && (
          <>
            <Link
              onClick={goBack}
              to={prompts[index - 1]}
              className="button"
              type="button"
            >
              Back
            </Link>
            <Link
              onClick={goForward}
              to={prompts[index + 1]}
              className="button"
              type="button"
            >
              Next
            </Link>
          </>
        )}
        {url === "/lesson-plan" && (
          <>
            <Link
              to={prompts[0]}
              className="button"
              type="button"
              onClick={goHome}
            >
              Home
            </Link>
            <Link to={prompts[5]} className="button" type="button">
              Retry
            </Link>
            <Link to={prompts[5]} className="button" type="button">
              Low-Resource
            </Link>
            <Link to={prompts[5]} className="button" type="button">
              Download
            </Link>
          </>
        )}
      </div>
    </div>
  );
}

export default Root;
