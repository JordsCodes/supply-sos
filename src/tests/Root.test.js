import React, { useState } from "react";
import Root from "../components/App";
import { render, screen, waitFor } from "@testing-library/react";

describe("Root", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<Root />);

    expect(asFragment()).toMatchSnapshot();
  });
  test("renders let's get started button when on landing page", async () => {
    render(<Root />);

    const startButton = screen.getByRole("link");

    expect(startButton).toHaveTextContent("Let's get started");
  });
  test("renders back and next buttons when start button clicked", async () => {});
});
