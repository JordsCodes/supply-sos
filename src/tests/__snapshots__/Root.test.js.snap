// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <h1
      class="heading"
    >
      SupplySoS
    </h1>
    <div
      id="prompt"
    >
      <div>
        <div
          style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
        >
          <div
            class="prompt"
          >
            <h2
              class="prompt_heading"
            >
              No planning? No problem.
            </h2>
            <h2
              class="prompt_heading_small"
            >
              No planning?
              <br />
               No problem.
            </h2>
            <p
              class="prompt_content"
            >
              Downloadable, customisable AI-generated lesson plans for Key Stages 1 and 2.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="navbar"
    >
      <a
        class="button"
        href="/year-group"
        type="button"
      >
        Let's get started
      </a>
    </div>
  </div>
</DocumentFragment>
`;
